spec/message.abnf

```diff
@@ -1,13 +1,16 @@
-message = [s] *(declaration [s]) body [s]
+message = pattern / complex-message
+
+complex-message = "{{" [s] *(declaration [s]) body [s] "}}"
 
 declaration = let s variable [s] "=" [s] expression
-body = pattern
+body = quoted-pattern
      / (selectors 1*([s] variant))
 
-pattern = "{" *(text / expression) "}"
-selectors = match 1*([s] expression)
-variant = when 1*(s key) [s] pattern
-key = literal / "*"
+quoted-pattern = "{{" pattern "}}"
+pattern        = *(text / expression)
+selectors      = match 1*([s] expression)
+variant        = when 1*(s key) [s] quoted-pattern
+key            = literal / "*"
```

Comment:
It would probably be easier to not align the `=` on adjacent lines, especially as we're looking to make further changes to the syntax.